### PR TITLE
Read a promotion feature

### PR DESCRIPTION
**Pull request by ChungLinWu0829**
### Description:
Added read-a-promotion route and tests

### Test result:
image

### Story reference: 
https://app.zenhub.com/workspaces/promotions-su24-001-666234565dda150276ad19c5/issues/gh/csci-ga-2820-su24-001/promotions/1

### CODE REVIEW

**Review by mingyi850 — APPROVED** [review]
Nice! LGTM, but consider using the datetime_utils functions to standardize datetime conversion for consistency

**Comment by mingyi850** [comment]
Oh also, could you run 'make lint' and help to fix the issues there - just to keep things clean for everyone's lint checks

File: tests/test_routes.py [hunk 1]
```diff
@@ -158,3 +160,30 @@ def mock_update_with_exception():
...
+        self.assertEqual(data["promotion_description"], existing_promotion.promotion_description)
+        self.assertEqual(data["promotion_type"], existing_promotion.promotion_type.name)
+        self.assertEqual(data["promotion_scope"], existing_promotion.promotion_scope.name)
+        self.assertEqual(data["start_date"], existing_promotion.start_date.isoformat())
+        self.assertEqual(data["end_date"], existing_promotion.end_date.isoformat())
+        self.assertEqual(data["promotion_value"], existing_promotion.promotion_value)
+        self.assertEqual(data["promotion_code"], existing_promotion.promotion_code)
+        self.assertEqual(UUID(data["created_by"]), existing_promotion.created_by)
+        self.assertEqual(UUID(data["modified_by"]), existing_promotion.modified_by)
+        self.assertEqual(data["created_when"], existing_promotion.created_when.isoformat())
```
[the remark this answers is not in the document]
Comment (ChungLinWu0829) [reply tests/test_routes.py]: Thank you for informing me of this, I had not noticed it.

**Comment by ChungLinWu0829** [comment]
> Oh also, could you run 'make lint' and help to fix the issues there - just to keep things clean for everyone's lint checks

Done. I've addressed nearly all linting issues except for one involving a broadly caught exception. I think we can create a new task labeled as 'technical debt' to refine our exception handling in future sprints.
image

Please let me know if you have any suggestions. If there are no further comments, I will merge the code today.

**Pull request merged**